src/include/daos_srv/bio.h

```diff
@@ -909,4 +909,60 @@ int bio_copy(struct bio_io_context *ioctxt, struct bio_sglist *bsgl_src,
 	     struct bio_sglist *bsgl_dst, unsigned int copy_size,
 	     struct bio_csum_desc *csum_desc);
 
+enum bio_mc_flags {
+	BIO_MC_FL_SYSDB		= (1UL << 0),	/* for sysdb */
+};
+
+/*
+ * Create Meta/Data/WAL blobs, format Meta & WAL blob
+ *
+ * \param[in]	xs_ctxt		Per-xstream NVMe context
+ * \param[in]	pool_id		Pool UUID
+ * \param[in]	meta_sz		Meta blob size in bytes
+ * \param[in]	wal_sz		WAL blob in bytes
+ * \param[in]	data_sz		Data blob in bytes
+ * \param[in]	csum_type	Checksum type used for WAL
+ * \param[in]	flags		bio_mc_flags
+ *
+ * \return			Zero on success, negative value on error.
+ */
+int bio_mc_create(struct bio_xs_context *xs_ctxt, uuid_t pool_id, uint64_t meta_sz,
+		  uint64_t wal_sz, uint64_t data_sz, uint16_t csum_type, enum bio_mc_flags flags);
+
+/*
+ * Destroy Meta/Data/WAL blobs
+ *
+ * \param[in]	xs_ctxt		Per-xstream NVMe context
+ * \param[in]	pool_id		Pool UUID
+ * \param[in]	flags		bio_mc_flags
+ *
+ * \return			Zero on success, negative value on error.
+ */
+int bio_mc_destroy(struct bio_xs_context *xs_ctxt, uuid_t pool_id, enum bio_mc_flags flags);
+
+/* Opaque meta context */
+struct bio_meta_context;
+
+/*
+ * Open Meta/Data/WAL blobs, load WAL header, alloc meta context
+ *
+ * \param[in]	xs_ctxt		Per-xstream NVMe context
+ * \param[in]	pool_id		Pool UUID
+ * \param[in]	umm		umem instance
+ * \param[out]	mc		BIO meta context
+ *
+ * \return			Zero on success, negative value on error
+ */
+int bio_mc_open(struct bio_xs_context *xs_ctxt, uuid_t pool_id, struct umem_instance *umm,
```

Comment:
maybe we need a skip_blob here to indicate there is no nvme usage, but we still need return a bio_meta_context to issue io to PMEM only.

see bio_ioctxt_open()

Reply:
Ok.